components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -54,124 +106,371 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
-  self.textLabel.text = text;
+    self.leadingButtonConstraintLeading = [self.leadingButton.leadingAnchor
+        constraintGreaterThanOrEqualToAnchor:self.buttonContainerView.leadingAnchor];
+    self.leadingButtonConstraintTop =
+        [self.leadingButton.topAnchor constraintEqualToAnchor:self.buttonContainerView.topAnchor];
+    self.leadingButtonConstraintTrailing = [self.leadingButton.trailingAnchor
+        constraintEqualToAnchor:self.buttonContainerView.trailingAnchor];
+    self.leadingButtonConstraintCenterY = [self.leadingButton.centerYAnchor
+        constraintEqualToAnchor:self.buttonContainerView.centerYAnchor];
+    self.leadingButtonConstraintTrailingWithTrailingButton =
+        [self.leadingButton.trailingAnchor constraintEqualToAnchor:self.trailingButton.leadingAnchor
+                                                          constant:-kButtonHorizontalIntervalSpace];
+    self.trailingButtonConstraintBottom = [self.trailingButton.bottomAnchor
+        constraintEqualToAnchor:self.buttonContainerView.bottomAnchor];
+    self.trailingButtonConstraintTop =
+        [self.trailingButton.topAnchor constraintEqualToAnchor:self.leadingButton.bottomAnchor
+                                                      constant:kButtonVerticalIntervalSpace];
+    self.trailingButtonConstraintTrailing = [self.trailingButton.trailingAnchor
+        constraintEqualToAnchor:self.buttonContainerView.trailingAnchor];
+  }
 }
 
-- (NSString *)text {
-  return self.textLabel.text;
+- (void)deactivateAllConstraints {
+  self.imageViewConstraintLeading.active = NO;
+  self.imageViewConstraintTopSmall.active = NO;
+  self.imageViewConstraintTopLarge.active = NO;
+  self.imageViewConstraintBottom.active = NO;
+  self.textLabelConstraintLeadingWithMargin.active = NO;
+  self.textLabelConstraintLeadingWithImage.active = NO;
+  self.textLabelConstraintTrailing.active = NO;
+  self.textLabelConstraintTop.active = NO;
+  self.textLabelConstraintCenterY.active = NO;
+  self.buttonContainerConstraintLeading.active = NO;
+  self.buttonContainerConstraintLeadingWithTextLabel.active = NO;
+  self.buttonContainerConstraintLeadingWithTextLabelGreater.active = NO;
+  self.buttonContainerConstraintTrailing.active = NO;
+  self.buttonContainerConstraintTopWithMargin.active = NO;
+  self.buttonContainerConstraintTopWithImageViewGreater.active = NO;
+  self.buttonContainerConstraintTopWithTextLabel.active = NO;
+  self.buttonContainerConstraintTopWithTextLabelGreater.active = NO;
+  self.buttonContainerConstraintBottom.active = NO;
+  self.leadingButtonConstraintLeading.active = NO;
+  self.leadingButtonConstraintTop.active = NO;
+  self.leadingButtonConstraintTrailing.active = NO;
+  self.leadingButtonConstraintCenterY.active = NO;
+  self.leadingButtonConstraintTrailingWithTrailingButton.active = NO;
+  self.trailingButtonConstraintBottom.active = NO;
+  self.trailingButtonConstraintTop.active = NO;
+  self.trailingButtonConstraintTrailing.active = NO;
```

Comment:
I'm nervous about this method getting out of sync if/when new constraints are added. I unfortunately don't see an immediately obvious solution to this problem though. I think this is a good area for future improvement in terms of the maintenance complexity of this code.